### PR TITLE
Naming convention 2.0.0

### PR DESCRIPTION
**Pull request by edouard-lopez**
:warning: To be merged with existing release `2.0.0` notes

---

## Migration Guide `v1.x` to `v2.0.0`

If you **customized your `pure`** using your `~/config.fish` you will have to rename some variables.

### :bomb: Base Color

* rename `$pure_color_white` as `$pure_color_light`

    ```diff
    -pure_color_white
    +pure_color_light
    ```

* rename `$pure_color_yellow` as `$pure_color_warning`

    ```diff
    -pure_color_yellow
    +pure_color_warning
    ```

* rename `$pure_color_red` as `$pure_color_danger`

    ```diff
    -pure_color_red
    +pure_color_danger
    ```
* rename `$pure_color_magenta` as `$pure_color_success`

    ```diff
    -pure_color_magenta
    +pure_color_success
    ```
* rename `$pure_color_gray` as `$pure_color_mute`

    ```diff
    -pure_color_gray
    +pure_color_mute
    ```
* rename `$pure_color_cyan` as `$pure_color_info`

    ```diff
    -pure_color_cyan
    +pure_color_info
    ```
* rename `$pure_color_blue` as `$pure_color_primary`

    ```diff
    -pure_color_blue
    +pure_color_primary
    ```

### :bomb: Prompt Colors Variables

* rename `$pure_color_symbol_error` as `$pure_color_prompt_on_error` 7de59a3

    ```diff
    -pure_color_symbol_error
    +pure_color_prompt_on_error
    ```

* rename `$pure_color_symbol_success` as `$pure_color_prompt_on_success` f94207b

    ```diff
    -pure_color_symbol_success
    +pure_color_prompt_on_success
    ```

### :bomb: Git Feature Variables

* rename `$pure_symbol_git_arrow_up` as `$pure_symbol_git_unpushed_commits`

    ```diff
    -pure_symbol_git_arrow_up
    +pure_symbol_git_unpushed_commits
    ```

* rename `$pure_symbol_git_arrow_down` as `$pure_symbol_git_unpulled_commits`

    ```diff
    -pure_symbol_git_arrow_down
    +pure_symbol_git_unpulled_commits
    ```

* split `$pure_color_git_pending_commits` as `$pure_color_git_unpushed_commits` and `$pure_color_git_unpulled_commits`

    ```diff
    -pure_color_git_pending_commits
    +pure_color_git_unpushed_commits
    +pure_color_git_unpulled_commits
    ```

### :bomb: Max Execution Time Feature Variables

* rename `$pure_command_max_exec_time` as `$pure_threshold_command_duration`

    ```diff
    -pure_command_max_exec_time
    +pure_threshold_command_duration
    ```

### :bomb: Title Bar Variable

* rename `$pure_symbol_horizontal_bar` as `$pure_symbol_title_bar_separator`

    ```diff
    -pure_symbol_horizontal_bar
    +pure_symbol_title_bar_separator
    ```

### :bomb: Beginning Of Prompt Variable

* rename `$pure_prompt_begin_with_current_directory` as `$pure_begin_prompt_with_current_directory`

    ```diff
    -pure_prompt_begin_with_current_directory
    +pure_begin_prompt_with_current_directory
    ```

### :bomb: SSH Variable

* rename `$pure_color_ssh_host` as `$pure_color_ssh_hostname`
    ```diff
    -pure_color_ssh_host
    +pure_color_ssh_hostname
    ```

### :bomb: Current Working Directory Variable

* rename `$pure_color_current_folder` as `$pure_color_current_directory`

    ```diff
    -pure_color_current_folder
    +pure_color_current_directory
    ```

### CODE REVIEW

File: conf.d/pure.fish [hunk 1]
```diff
@@ -15,6 +15,7 @@ _pure_set_default pure_color_prompt_on_success $pure_color_success
 _pure_set_default pure_symbol_git_unpulled_commits "⇣"
 _pure_set_default pure_symbol_git_unpushed_commits "⇡"
 _pure_set_default pure_symbol_git_dirty "*"
+_pure_set_default pure_color_git_pending_commits $pure_color_info
```
Comment (schrodincat) [inline conf.d/pure.fish]: I still think we should use two variables🖖 I thought @rafaelrinaldi agreed with me on this: https://github.com/rafaelrinaldi/pure/issues/130#issuecomment-453797592

**Comment by edouard-lopez** [comment]
@filipebarros thanks, I messed-up the first section.

I remove the commit hash as they will be incorrect after the merge/rebase.

**Comment by edouard-lopez** [comment]
@schrodincat My bad, I split the variable, updated the tests and rename the file see 7f24374

**Comment by edouard-lopez** [comment]
@rafaelrinaldi do you wanna do a pair-review session?

**Comment by schrodincat** [comment]
I'm also waiting for this to be merged

**Comment by rafaelrinaldi** [comment]
@edouard-lopez Let's do it. Ping me on the IM whenever you free up some time.

**Comment by edouard-lopez** [comment]
@schrodincat I understand you might be frustrated by the delay. However, this kind of massive refactor isn't something we will do often.

The simplicity of [Trunk-based development](https://trunkbaseddevelopment.com/) is really effective with no overhead. It helps make smaller and simpler changes with:
* fewer merge conflicts, 
* easier code review, 
* encourage PR (require CI) and discussion/pair-programming
* and faster deployment. 
Of course, for this to work we need strong test culture and PR should not be merged is they lack sufficient tests to prove they are functional.

@rafaelrinaldi good I will ping you in the afternoon GMT

**Comment by schrodincat** [comment]
@edouard-lopez my only concern is about possible mismatch between actual version and version in `pure_version`.

For example if we would merge this PR with master, and user install it, `pure_version` would report 2.0.0, however it’s not 2.0.0, since it doesn’t contain other features planned for 2.0.0.

Essentially current approach requires making release and bumping `pure_version` with **every** merge in master. This is tedious and error-prone, unless it can be automated somehow.

**Comment by edouard-lopez** [comment]
Thanks @schrodincat, let's see how often this happen and decide what strategy to adopt.

Moreover this is the last PR before release of `2.0.0`, see #74.

**Comment by edouard-lopez** [comment]
As discuss in our telegram channel:

> We will add a _tools/migrate-to-2.0.0.fish_ script in order to **help users do the migration**: running a series of sed command to rename the variable.
>
> For the end user to **update their fish config** by providing the file to parse/update
> 
> this mechanism can definitely be helpful in the future, if we’ll have more changes.
> 
> some **variable only changed names while other also changed type** (_e.g._ from integer to boolean). So we will need special rules to update them when the option is `0` or `1`

**Comment by rafaelrinaldi** [comment]
Looks great to me. Ship it!

**Pull request merged**